controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1346,12 +1352,14 @@ private TxnStatus handleDataNotFoundException(Throwable ex) {
                     // mark transaction as committed in metadata store.
                     .thenCompose(x -> commitTransaction(txnId)
                             .thenAccept(done -> {
+                                DYNAMIC_LOGGER.incCounterValue(nameFromStream(COMMIT_TRANSACTION, scope, name), 1);
                                 log.debug("transaction {} on stream {}/{} committed successfully", txnId, scope, name);
                             }));
         }
-        return future
-                .thenCompose(x -> Futures.toVoid(updateCommittingTxnRecord(new Data(CommittingTransactionsRecord.EMPTY.toBytes(),
-                        record.getVersion()))));
+        return future.thenCompose(x -> Futures.toVoid(updateCommittingTxnRecord(new Data(CommittingTransactionsRecord.EMPTY.toBytes(),
+                             record.getVersion()))))
+                     .thenRun(() -> getNumberOfOngoingTransactions().thenAccept(count ->
```

Comment:
as discussed offline, doing this "expensive" computation of getNumberOfOngoingTransactions at the end of each individual commit operation is costly. 
we should instead do this in CommitRequestHandler. CommitRequestHandler fetches a batch of available transactions that have been marked for commit and commits them one at a time. We could do this metrics reporting at the end of commit workflow execution so that the expensive computation happens once for a batch of transactions.

Reply:
I think that `OPEN_TRANSACTIONS` is composed outside of the loop, so it will be executed only once. Anyway, as per out offline discussion, I will move the logging code to `AbstractStreamMetadataStore` (`OPEN_TRANSACTIONS`) and `CommitRequestHandler` (`COMMIT_TRANSACTION`) to avoid `PersistentStreamBase` to perform metrics logging as well.